src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/_params.py

```diff
@@ -103,3 +104,9 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs): # pylin
 
 register_cli_argument('appservice web config hostname', 'webapp', help="webapp name", completer=get_resource_name_completion_list('Microsoft.Web/sites'), id_part='name')
 register_cli_argument('appservice web config hostname', 'name', arg_type=name_arg_type, completer=get_hostname_completion_list, help="hostname assigned to the site, such as custom domains", id_part='child_name')
+
+register_cli_argument('appservice web source-control', 'is_manual_integration', action='store_true', help='disable automatic sync between source control and web')
+register_cli_argument('appservice web source-control', 'repo_url', help='repository url to pull the latest source from, e.g. https://github.com/foo/foo-web')
+register_cli_argument('appservice web source-control', 'branch', help='the branch name of the repository')
+register_cli_argument('appservice web source-control', 'is_mercurial', action='store_true', help='is Mercurial repository(not Git)')
```

Comment:
Couldn't we expose this is a type parameter with git as the default? If they ever add support for say, SVN, then this flag-based approach won't work and they'd have to create some kind of enum. If we make the CLI that way to start then if the list does expand we don't have to introduce a breaking change into the CLI because we'd just be adding to the supported list of options.